### PR TITLE
Automatically serve API on https with self-signed cert if no cert is provided

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

File: cmd/kube-apiserver/apiserver.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ var (
 		"The port from which to serve read-only resources. If 0, don't serve on a "+
 		"read-only address. It is assumed that firewall rules are set up such that "+
 		"this port is not reachable from outside of the cluster.")
-	securePort              = flag.Int("secure_port", 0, "The port from which to serve HTTPS with authentication and authorization. If 0, don't serve HTTPS ")
+	securePort              = flag.Int("secure_port", 8443, "The port from which to serve HTTPS with authentication and authorization. If 0, don't serve HTTPS ")
 	tlsCertFile             = flag.String("tls_cert_file", "", "File containing x509 Certificate for HTTPS.  (CA cert, if any, concatenated after server cert).")
```
Comment (jbeda) [inline cmd/kube-apiserver/apiserver.go]: Add a comment here that one will be created if not set here and saved into xyz directory.

File: pkg/util/crypto.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	} else {
+		template.DNSNames = append(template.DNSNames, host)
+	}
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &priv.PublicKey, priv)
+	if err != nil {
+		return
+	}
+
+	certOut, err := ioutil.TempFile("", "cert.pem")
```
Comment (jbeda) [inline pkg/util/crypto.go]: Putting this in `/tmp` seems like it might be a bad idea.  How about `/var/run/kubernetes` instead?

Comment (liggitt) [inline pkg/util/crypto.go]: works for me... is there a way to get the preferred `var` location to use?

Comment (jbeda) [inline pkg/util/crypto.go]: Ug - I don't know.  We could make that a flag. `/var/run/kubernetes` probably isn't a bad place to start with as it is part of LSB.

Comment (liggitt) [inline pkg/util/crypto.go]: For now, I'll just pass in the paths... we can make configurable later if desired

**Comment by liggitt** [comment]
@jbeda PTAL

**Pull request merged**